tests/fixtures/get-test-fixture.ts

```diff
@@ -1,13 +1,17 @@
-import { ExecutionContext } from "ava"
+import type { AnySoupElement } from "@tscircuit/soup"
+import type { ExecutionContext } from "ava"
 import { createProjectBuilder } from "../../src"
 import { logLayout } from "../utils/log-layout"
+import { schematicSnapshotOutput } from "./schematic-snapshot-output"
 
 /**
  * Consolidates common test functions and components for simpler test definition
  */
 export const getTestFixture = (t: ExecutionContext) => {
   return {
-    logSoup: (soup: any) => logLayout(t.title, soup),
+    logSoup: (soup: AnySoupElement[]) => logLayout(t.title, soup),
     pb: createProjectBuilder(),
+    schematicSnapshot: (soup: AnySoupElement[]) =>
+      schematicSnapshotOutput(soup),
```

Comment:
We should find the path to the test directory, then these can be committed as snapshots.

Note: For `jest`-style snapshots (which might be a good idea if things are messy) we could use `${myTest.testDir}/__snapshots__/${t.title}.png`

```suggestion
      schematicSnapshotOutput(`${myTest.testDir}/${t.title}.png`, soup),
```

Reply:
I thought about taking the path parameter, but thinking from the point of view, that this is just a debug tool. Used it to generate at the home directory

Reply:
@imrishabh18 if it's a snapshot (like snapshot testing) we should commit the result so that we can see the diff on PRs. You won't need to provide a path as long as we automatically use `t.title`

Note: We should detect if the `-u` flag is provided in the future, and if the file has changed (using looks-same with a threshold) we throw UNLESS the `-u` flag is set (indicating the snapshot should intentionally be updated)

Reply:
I think `debug.tscircuit.com` should be the preferred mechanism for debug testing (we can improve it more e.g. fix the dark mode issues, fix the "everyone basically edits the global data" etc.). These snapshots will be insanely valuable for "proving the output looks like I said it does"

Reply:
For committing the result and see the difference between the snapshots, I think we can use this npm package which checks the difference between the two [looks-same]( https://www.npmjs.com/package/looks-same). Will add this in another PR.

Reply:
for sure, yea future work!!